torch/csrc/jit/frontend/tree_views.cpp

```diff
@@ -0,0 +1,40 @@
+#include <torch/csrc/jit/frontend/tree_views.h>
+
+namespace torch {
+namespace jit {
+
+namespace {
+void collectUnresolvedNames(
+    std::vector<std::string>& names,
+    const TreeView& node) {
+  if (node.kind() == TK_ASSIGN) {
+    for (const auto& expr : Assign{node.get()}.lhs_list()) {
+      collectUnresolvedNames(names, expr);
+    }
+  } else if (node.kind() == TK_TUPLE_LITERAL) {
+    for (const auto& expr : TupleLiteral{node.get()}.inputs()) {
+      collectUnresolvedNames(names, expr);
+    }
+  } else if (node.kind() == TK_LIST_LITERAL) {
+    for (const auto& expr : ListLiteral{node.get()}.inputs()) {
+      collectUnresolvedNames(names, expr);
+    }
```

Comment:
Nice! Could you add test case covering these branches as well? including nested tuple literals and list literals? Thanks!